### ISSUE TITLE
MEMORY: rep(NA, ...) -> rep(NA_character_, ...)

### ISSUE THREAD

**Issue by HenrikBengtsson** [Bioconductor/Biobase#3]
In master branch (commit 073511c) there are a few `rep(NA, ...)` that looks like they could (=should) be `rep(NA_character_, ...)` in order to avoid unecessary coercion from logical to character;
```
$ grep -F "rep(NA" R/*.R
R/methods-AnnotatedDataFrame.R:            varMetadata <- data.frame(labelDescription = rep(NA, ncol(data)))
R/methods-AnnotatedDataFrame.R:                    rep(NA, nrow(varMetadata))
R/methods-AnnotatedDataFrame.R:    varMetadata[["labelDescription"]] <- as.character(rep(NA, nrow(varLabels)))
R/methods-AnnotatedDataFrame.R:    varMetadata <- data.frame(labelDescription = rep(NA, ncol(data)))
R/methods-AnnotatedDataFrame.R:        varMetadata[["labelDescription"]] <- rep(NA, nrow(varMetadata))
```

**Comment by hpages** [comment]
Even better: use `rep.int` instead of `rep`. The former tends to be slightly faster (around 8% in this case but can be more for other atomic types).

**Comment by mtmorgan** [comment]
OK @Kayla-Morrell can you in devel only create a branch and

- Replace `rep(NA, ...)` with `rep.int(NA_character_, ...)`.
- In the data.frame constructor in the first and fourth line add `stringsAsFactors = FALSE`.
- Version bump
- build & check

Create a merge request with me as reviewer when done. Thanks!

**Comment by mtmorgan** [comment]
Closed with 3ba68613cff60733619fd522572d8e989ac72dfd
